### PR TITLE
Use instruction coverage plugin

### LINKED ISSUE #978: move coverage calculation to plugin

### PR DESCRIPTION
**Pull request by JoranHonig**
Removes coverage related logic from core svm

Extends the instruction coverage plugin with functionality to report coverage increase between transactions

closes #978

### CODE REVIEW

File: mythril/laser/ethereum/plugins/implementations/coverage.py [hunk 1]
```diff
@@ -47,13 +54,36 @@ def execute_state_hook(global_state: GlobalState):
...
+
+        @symbolic_vm.laser_hook("stop_sym_trans")
+        def execute_stop_sym_trans_hook():
+            end_coverage = self._get_covered_instructions()
+            log.info(
+                "Number of new instructions covered in tx %d: %d"
+                % (self.tx_id, end_coverage - self.initial_coverage)
+            )
+            self.tx_id += 1
+
```
Comment (norhh) [inline mythril/laser/ethereum/plugins/implementations/coverage.py]: When will this function get executed?

File: mythril/laser/ethereum/plugins/implementations/coverage.py [hunk 2]
```diff
@@ -47,13 +54,36 @@ def execute_state_hook(global_state: GlobalState):
...
+    def _get_covered_instructions(self) -> int:
```
Comment (norhh) [inline mythril/laser/ethereum/plugins/implementations/coverage.py]: When will this function get executed?

Comment (JoranHonig) [reply mythril/laser/ethereum/plugins/implementations/coverage.py]: In the two previous hooks, whom are executed at each transaction end & start

File: mythril/laser/ethereum/svm.py [hunk 3]
```diff
@@ -527,6 +490,10 @@ def register_laser_hooks(self, hook_type: str, hook: Callable):
             self._start_sym_exec_hooks.append(hook)
         elif hook_type == "stop_sym_exec":
             self._stop_sym_exec_hooks.append(hook)
+        elif hook_type == "start_sym_trans":
+            self._start_sym_trans_hooks.append(hook)
+        elif hook_type == "stop_sym_trans":
```
Comment (norhh) [inline mythril/laser/ethereum/svm.py]: As this part might  increase in the future, we can make a static dictionary of strings and functions and directly use it, so each hook access would be in O(1) if we move the static dictionary into the svm initialization. but it is a negligible optimization, anyway using a dictionary in these cases is a bit more clean, WDYT?

Comment (JoranHonig) [reply mythril/laser/ethereum/svm.py]: I agree, was planning on doing something similar. But not in this pr 😅

**Pull request merged**